extension/user/login.js

```diff
@@ -0,0 +1,53 @@
+const crypto = require('crypto')
+const {EINVALIDCREDENTIALS, ENOTFOUND, customError} = require('../error')
+/**
+ * @typedef {Object} LoginInputArgs
+ * @property {string} strategy
+ * @property {Object} parameters
+ * @property {string} parameters.login
+ * @property {string} parameters.password
+ */
+
+/**
+ * @param {SDKContext} context
+ * @param {LoginInputArgs} input
+ * @param {function} cb
+ */
+module.exports = function (context, input, cb) {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
+      userId: context.meta.userId
+    })
+  }
+
+  context.storage.extension.get(input.parameters.login, (errUserId, userId) => {
+    if (errUserId) {
+      return cb(errUserId)
+    }
+
+    if (!userId) {
+      return cb(customError(ENOTFOUND, 'User is not found'))
+    }
+
+    context.storage.extension.get(userId, (err, user) => {
+      if (err) {
+        return cb(err)
+      }
+
+      if (!user) {
+        return cb(customError(ENOTFOUND, 'User not found'))
+      }
+
+      // Check password match
+      const password = crypto.createHash('md5').update(input.parameters.password).digest('hex')
```

Comment:
Are we reading the cake user? If not, we should not use md5 anymore.
*EDIT:* Okay, we're reading from cake database. Need to clearify if this is expected, since the data was supposed to be "not from legacy system".